freud/box.pyx

```diff
@@ -422,6 +422,104 @@ cdef class Box:
         self.thisptr.center(<vec3[float]*> &l_points[0, 0], Np, l_masses_ptr)
         return vecs
 
+    def compute_distances(
+            self, query_points, points, query_point_indices, point_indices):
+        R"""Calculate distances between two sets of points, using periodic boundaries.
+
+        Args:
+            query_points (:math:`\left(N, 3, \right)` :class:`numpy.ndarray`):
+                Array of query points.
+            points (:math:`\left(N, 3, \right)` :class:`numpy.ndarray`):
+                Array of points.
+            query_points_indices (:math:`\left(N, \right)` :class:`numpy.ndarray`):
+                Array of indices that correspond to a set of query points.
+            points_indices (:math:`\left(N, \right)` :class:`numpy.ndarray`):
+                Array of indices that correspond to a set of points, of same length and query_points_indices.
+
+        Returns:
+            :math:`\left(N, \right)` :class:`numpy.ndarray`:
+                Array of distances between query points and points.
+        """   # noqa: E501
+        query_points = np.asarray(query_points)
+        points = np.asarray(points)
+        query_point_indices = np.atleast_1d(query_point_indices)
```

Comment:
I did this in case someone tries to use specify indicies as an integer instead of an array. I will remove this if we want to discourage this kind of input.

Reply:
I think if we follow my API suggestion on the other comment (getting rid of indices), we should do an `atleast_2d` on the `points` and `query_points` to allow single point calculation.